src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_mask.c

```diff
@@ -1,59 +1,808 @@
+    SET_MOON_MASK_BIT(2, 0x4),  // SLOT_MASK_SCENTS
+    SET_MOON_MASK_BIT(2, 0x20), // SLOT_MASK_GORON
+    SET_MOON_MASK_BIT(0, 0x40), // SLOT_MASK_ROMANI
+    SET_MOON_MASK_BIT(0, 0x80), // SLOT_MASK_CIRCUS_LEADER
+    SET_MOON_MASK_BIT(0, 0x2),  // SLOT_MASK_KAFEIS_MASK
+    SET_MOON_MASK_BIT(1, 0x2),  // SLOT_MASK_COUPLE
+    SET_MOON_MASK_BIT(0, 0x1),  // SLOT_MASK_TRUTH
+    SET_MOON_MASK_BIT(2, 0x40), // SLOT_MASK_ZORA
+    SET_MOON_MASK_BIT(1, 0x20), // SLOT_MASK_KAMARO
+    SET_MOON_MASK_BIT(1, 0x8),  // SLOT_MASK_GIBDO
+    SET_MOON_MASK_BIT(0, 0x20), // SLOT_MASK_GARO
+    SET_MOON_MASK_BIT(1, 0x40), // SLOT_MASK_CAPTAIN
+    SET_MOON_MASK_BIT(2, 0x8),  // SLOT_MASK_GIANT
+    SET_MOON_MASK_BIT(2, 0X80), // SLOT_MASK_FIERCE_DEITY
 };
 
-u8 D_8082B6CC[] = { 0x35, 0x33, 0x34, 0x32, 0xFF };
+s16 sMaskMagicArrowEffectsR[] = { 255, 100, 255 };
+s16 sMaskMagicArrowEffectsG[] = { 0, 100, 255 };
+s16 sMaskMagicArrowEffectsB[] = { 0, 255, 100 };
 
-s16 D_8082B6D4[] = {
-    0x0294,
-    0x0384,
-    0x0474,
-    0x0000,
-};
+void KaleidoScope_DrawMaskSelect(PlayState* play) {
+    PauseContext* pauseCtx = &play->pauseCtx;
+    u16 i;
+    u16 j;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    KaleidoScope_SetCursorVtx(pauseCtx, pauseCtx->cursorSlot[PAUSE_MASK] * 4, pauseCtx->maskVtx);
+
+    func_8012C8AC(play->state.gfxCtx);
+
+    // Draw a white box around the items that are equipped on the C buttons
+    // Loop over c-buttons (i) and vtx offset (j)
+    gDPSetCombineMode(POLY_OPA_DISP++, G_CC_MODULATEIA_PRIM, G_CC_MODULATEIA_PRIM);
+    gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, pauseCtx->alpha);
+    for (i = 0, j = NUM_MASK_SLOTS * 4; i < 3; i++, j += 4) {
+        if (GET_CUR_FORM_BTN_ITEM(i + 1) != ITEM_NONE) {
+            if (GET_CUR_FORM_BTN_SLOT(i + 1) >= NUM_ITEM_SLOTS) {
+                gSPVertex(POLY_OPA_DISP++, &pauseCtx->maskVtx[j], 4, 0);
+                POLY_OPA_DISP = func_8010DC58(POLY_OPA_DISP, gEquippedItemOutlineTex, 32, 32, 0);
+            }
+        }
+    }
+
+    gDPPipeSync(POLY_OPA_DISP++);
+
+    // Draw the item icons
+    // Loop over slots (i) and vtx offset (j)
+    gDPSetCombineMode(POLY_OPA_DISP++, G_CC_MODULATEIA_PRIM, G_CC_MODULATEIA_PRIM);
+    for (j = 0, i = 0; i < NUM_MASK_SLOTS; i++, j += 4) {
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, pauseCtx->alpha);
 
-s16 D_8082B6DC[] = {
-    0x044C,
-    0x0398,
-    0x044C,
-    0x0000,
+        if (((void)0, gSaveContext.save.inventory.items[i + NUM_ITEM_SLOTS]) != ITEM_NONE) {
+            if (!CHECK_MOON_MASK_TAKEN(i)) {
+                if ((pauseCtx->unk_200 == 0) && (pauseCtx->pageIndex == PAUSE_MASK) &&
+                    (pauseCtx->cursorSpecialPos == 0) &&
+                    gMaskPlayerFormSlotRestrictions[(void)0, gSaveContext.save.playerForm][i]) {
+                    if ((sMaskEquipState == EQUIP_STATE_MAGIC_ARROW_HOVER_OVER_BOW_SLOT) && (i == SLOT_ARROW_ICE)) {
```

Comment:
Is this code actually used, or is it literally just a leftover from copypaste?

Reply:
The second part is still used, the `// Increase the size of the selected item`, but pretty sure this block with the magic arrow isn't used here (or even used in the item page). It's weird

Reply:
Yeah, I feel like we need a note somewhere saying that most of this file is identical to the items page. I wish there was a good way to show the differences.